validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/DeleteKeys.java

```diff
@@ -49,14 +47,21 @@ public DeleteKeys() {
             .response(SC_OK, "Success response", ValidatorTypes.DELETE_KEYS_RESPONSE_TYPE)
             .withAuthenticationResponses()
             .build());
+
+    this.keyManager = keyManager;
   }
 
   @Override
   public void handle(final RestApiRequest request) throws JsonProcessingException {
-    List<BLSPublicKey> keys = request.getRequestBody();
-    for (BLSPublicKey key : keys) {
-      LOG.debug("Delete key: {}", key.toBytesCompressed().toShortHexString());
+    try {
+      DeleteKeysRequest deleteRequest = request.getRequestBody();
+      if (deleteRequest == null) {
+        request.respondError(SC_BAD_REQUEST, "Could not read request body");
```

Comment:
hmm, feels like we shouldn't need this null check and it should just be handled by the framework.  I'd say if it gets null it should throw `JsonProcessingException` automatically.  In the future we may need to add a way to mark the request body as optional but I think that's a pretty rare case.

Reply:
this is one of those small comments that got big, but its fixed now :)